File: dotnet/private/rules/common/nunit/shim.cs
internal static class NUnitShim {
    public static int Main(string[] args) {
        var argsList = new System.Collections.Generic.List<string>(args);

        // XML test output: when Bazel sets XML_OUTPUT_FILE, write NUnit3 results
        var xmlOutputFile = System.Environment.GetEnvironmentVariable("XML_OUTPUT_FILE");
        if (xmlOutputFile != null) {
            argsList.Add("--result=" + xmlOutputFile + ";format=nunit3");
        }

        // Test sharding: TEST_SHARD_STATUS_FILE touch is handled in launcher.
        // NUnitLite doesn't support modulo-based test filtering natively;
        // Bazel sharding primarily splits across test targets.

        return new NUnitLite.AutoRun().Execute(argsList.ToArray());
    }
}

File: dotnet/private/rules/common/nunit/shim.fs
[<EntryPoint>]
let main argv =
    let argsList = System.Collections.Generic.List<string>(argv)

    // XML test output: when Bazel sets XML_OUTPUT_FILE, write NUnit3 results
    let xmlOutputFile = System.Environment.GetEnvironmentVariable("XML_OUTPUT_FILE")
    if xmlOutputFile <> null then
        argsList.Add("--result=" + xmlOutputFile + ";format=nunit3")

    // Test sharding: TEST_SHARD_STATUS_FILE touch is handled in launcher.
    NUnitLite.AutoRun().Execute(argsList.ToArray())
